spec/integration/account_deletion_spec.rb

```diff
@@ -1,128 +1,46 @@
-      end
-
-      # notifications
-      @notifications = []
-      3.times do
-        @notifications << FactoryGirl.create(:notification, recipient: bob)
-      end
-
-      # services
-      @services = []
-      3.times do
-        @services << FactoryGirl.create(:service, user: bob)
-      end
-
-      # block
-      @block = bob.blocks.create!(person: eve.person)
-
-      AccountDeleter.new(bob.person.diaspora_handle).perform!
-      bob.reload
-    end
-
-    it "deletes all of the user's preferences" do
-      expect(UserPreference.where(id: @prefs.map(&:id))).to be_empty
-    end
-
-    it "deletes all of the user's notifications" do
-      expect(Notification.where(id: @notifications.map(&:id))).to be_empty
-    end
-
-    it "deletes all of the users's blocked users" do
-      expect(Block.where(id: @block.id)).to be_empty
-    end
-
-    it "deletes all of the user's services" do
-      expect(Service.where(id: @services.map(&:id))).to be_empty
-    end
-
-    it "deletes all of bobs share visiblites" do
-      expect(ShareVisibility.where(id: @users_sv.map(&:id))).to be_empty
-      expect(ShareVisibility.where(id: @persons_sv.map(&:id))).to be_empty
-    end
-
-    it "deletes all of bobs aspect visiblites" do
-      expect(AspectVisibility.where(id: @aspect_vis.map(&:id))).to be_empty
-    end
+describe "deleteing account", type: :request do
+  def account_removal_method
+    AccountDeleter.new(subject.diaspora_handle).perform!
+    subject.reload
+  end
 
-    it "deletes all aspects" do
-      expect(bob.aspects).to be_empty
-    end
+  context "of local user" do
+    subject(:user) { FactoryGirl.create(:user_with_aspect) }
 
-    it "deletes all user contacts" do
-      expect(bob.contacts).to be_empty
-    end
-
-    it "clears the account fields" do
-      bob.send(:clearable_fields).each do |field|
-        expect(bob.reload[field]).to be_blank
-      end
+    before do
+      DataGenerator.create(subject, :generic_user_data)
+    end
+
+    it "deletes all of the user data" do
+      expect {
+        account_removal_method
+      }.to change(nil, "user preferences empty?") { UserPreference.where(user_id: user.id).empty? }.to(be_truthy)
+        .and(change(nil, "notifications empty?") { Notification.where(recipient_id: user.id).empty? }.to(be_truthy))
+        .and(change(nil, "blocks empty?") { Block.where(user_id: user.id).empty? }.to(be_truthy))
+        .and(change(nil, "services empty?") { Service.where(user_id: user.id).empty? }.to(be_truthy))
+        .and(change(nil, "share visibilities empty?") { ShareVisibility.where(user_id: user.id).empty? }.to(be_truthy))
+        .and(change(nil, "aspects empty?") { user.aspects.empty? }.to(be_truthy))
+        .and(change(nil, "contacts empty?") { user.contacts.empty? }.to(be_truthy))
+        .and(change(nil, "tag followings empty?") { user.tag_followings.empty? }.to(be_truthy))
+        .and(change(nil, "clearable fields blank?") {
```

Comment:
This and another message of rubocop is inadequate and maybe errorneous. I don't know how to fix it, but maybe will come with some idea later.